### PR TITLE
Add CellBottomView

### PR DESCRIPTION
**Pull request by m-sugawara**
## Overview
- add CellBottomView for customizing message
- add an example for showing sample usage of CellBottomView

## Screenshot
Simulator Screen Shot - iPhone Xʀ - 2019-09-03 at 19 49 59

### CODE REVIEW

**Comment by m-sugawara** [comment]
I will merge now. Please check if you have time 🙏  @quipper/aya-native-ios-devs

File: Example/Sources/Base.lproj/LaunchScreen.storyboard [hunk 1]
```diff
@@ -22,7 +22,7 @@
                         <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                         <subviews>
                             <imageView userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="mklogo" translatesAutoresizingMaskIntoConstraints="NO" id="NHX-yN-7Jk">
-                                <rect key="frame" x="76" y="99" width="222" height="127"/>
+                                <rect key="frame" x="76.5" y="99" width="222" height="127.5"/>
```
Comment (manicmaniac) [inline Example/Sources/Base.lproj/LaunchScreen.storyboard:25]: Maybe a nitpick and not sure but I guess this indicates the rect is misaligned.
Open a simulator and tap `Debug` → `Color Misaligned Images` on the menu bar would tell you if it's misaligned.
Misaligned image itself is not a problem but causes performance losses on particular devices.

File: Example/Sources/Base.lproj/LaunchScreen.storyboard [hunk 2]
```diff
@@ -42,6 +42,6 @@
         </scene>
     </scenes>
     <resources>
-        <image name="mklogo" width="444" height="127"/>
+        <image name="mklogo" width="444" height="127.5"/>
```
Comment (manicmaniac) [inline Example/Sources/Base.lproj/LaunchScreen.storyboard:45]: Same as above.

File: Example/Sources/View Controllers/AnnouncementExampleViewController.swift [hunk 3]
```diff
@@ -17,13 +17,28 @@ class AnnouncementExampleViewController: MessagesViewController {
...
-        initializeUI()
+
+        configureMessageCollectionView()
         conversationMessages = ConversationMessageMockFactory.getConversations()
     }
 
-    private func initializeUI() {
+    func configureMessageCollectionView() {
+
+        let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout
```
Comment (manicmaniac) [inline Example/Sources/View Controllers/AnnouncementExampleViewController.swift:33]: Wanna use `if let` to reduce `nil` checks?

File: Sources/Protocols/MessagesDisplayDelegate.swift [hunk 4]
```diff
@@ -71,7 +71,19 @@ public protocol MessagesDisplayDelegate: AnyObject {
...
+    /// Used to configure the `CellBottomView` in a `MessageContentCell` class.
+    ///
+    /// - Parameters:
+    ///   - avatarView: The `CellBottomView` of the cell.
+    ///   - message: The `MessageType` that will be displayed by this cell.
+    ///   - indexPath: The `IndexPath` of the cell.
+    ///   - messagesCollectionView: The `MessagesCollectionView` in which this cell will be displayed.
+    ///
+    /// - Note:
+    ///   The default image configured by this method is `?`.
```
Comment (manicmaniac) [inline Sources/Protocols/MessagesDisplayDelegate.swift:84]: :+1: for a descriptive comment.

**Review by manicmaniac — COMMENTED** [review]
LGTM :+1:

File: Example/Sources/Base.lproj/LaunchScreen.storyboard (hunk 1 above)
Comment (m-sugawara) [reply Example/Sources/Base.lproj/LaunchScreen.storyboard:25]: Thank you for sharing! I will check 👍

**Pull request merged**